### DISCUSSION TITLE
反応しないキーがあります

### DISCUSSION THREAD

**Discussion by konpuku** [Yowkees/keyball#57] [Q&A] [answered]
ビルドガイド[3-6．ProMicoの書き込みと動作確認]の、テスト用ファームウェアの書き込み後の動作確認で躓いています。

・左手側一番左の列の最下段
・左手側一番右の列の最下段(TRRSソケットの真下)
の2キーが反応しません。

ダイオードの導通確認は実施済です。
トラックボールは右手側で、左手側の動作確認には「keyball_rev1_noball_test.hex」を使用しています。

どの部分を確認すればよいでしょうか。

**Comment by Yowkees** [comment, marked as answer]
回路図を公開しなければと思いながら、なかなか出来ていませんでしたが、本日公開できました。
下記リポジトリにKeyball46のトラックボール無し側基板の回路図(PDF)を入れました。

https://github.com/Yowkees/keyball/blob/main/keyball46/doc/rev1/Keyball46_Circuit_NoBallSide.pdf

さて、お問い合わせいただいた問題ですが一度ProMicroを取り外していただき、ProMicroを差し入れるスルーホールのCOL0とROW3にテスターを当てます（COL0に赤い端子を当ててください）。この状態で『左手側一番左の列の最下段キー』を押したときに導通するかご確認ください。
また、『左手側一番右の列の最下段(TRRSソケットの真下)』についてはCOL5とROW3にテスターを当ててご確認ください。

スイッチを押しても変化しない場合、PCBソケットとダイオードのはんだ付けをご確認ください。

お手数ですが動作確認よろしくお願いいたします。

**Reply by konpuku** [reply]
ご回答いただきありがとうございます。また、回路図も公開いただきありがとうございます。
いただいた手順で確認したところ、
『左手側一番右の列の最下段(TRRSソケットの真下)』については導通し、
『左手側一番左の列の最下段キー』については導通しませんでした。
前者については、なぜ導通したのかわからなったのですが
(3-6から先に進めて、左右判定ジャンパをしたところ反応しました)、
後者については、ダイオードのはんだ付けを再度確認したところ、確認するダイオードを誤っておりました。
(キーの直上のものを確認していましたが、裏面からみて左隣のキーの左のものを確認すべきでした)
そちらのダイオードが逆だったのを修正することで、全キーの反応が確認できました。
こちらの確認不足でお手間を取らせてしまい申し訳ございませんでした。
